src/model/paths_types.ml

```diff
@@ -620,6 +629,7 @@ module rec Reference : sig
     [ `Resolved of Resolved_reference.field_parent
     | `Root of string * tag_parent
     | `Dot of label_parent * string
+    | `Module_path of hierarchy
```

Comment:
Module path can only be of the form: `/libname/`. For example: `{!/libname/module-M.type-t}`.

I think that could be reflected in the type?

Reply:
I guess that https://github.com/ocaml/odoc/pull/1142#discussion_r1672366269 answers no to the question!

Reply:
I think this can happen with `{!/M.type-t}`. I'm not in favor of enforcing that in the parser. The resolver has to match on the path to be able to lookup `libname`, that's where the reference should be rejected.